### ISSUE TITLE
Epoch loss not calculated properly

### ISSUE THREAD

**Issue by Lm36** [nicknochnack/PyTorchin15#4]
The "Epoch loss" being printed is only the loss for the last batch in the epoch, not the loss for the full epoch.

Issue:
```python3
    for epoch in range(10): 
        for batch in dataset: 
            X,y = batch 
            X, y = X.to('cuda'), y.to('cuda') 
            yhat = clf(X) 
            loss = loss_fn(yhat, y) # loss is only assigned to the current batch

            # Apply backprop 
            opt.zero_grad()
            loss.backward() 
            opt.step() 

        print(f"Epoch:{epoch} loss is {loss.item()}") # loss of last batch printed as epoch loss
```

Solution:
```python3
    for epoch in range(10): 
        running_loss = 0.0
        for batch in dataset: 
            X,y = batch 
            X, y = X.to('cuda'), y.to('cuda') 
            yhat = clf(X) 
            loss = loss_fn(yhat, y) 
            running_loss += loss.item() # Add batch loss to epoch running loss

            # Apply backprop 
            opt.zero_grad()
            loss.backward() 
            opt.step() 

        epoch_loss = running_loss/len(dataset) # Calculate loss for the epoch
        print(f"Epoch:{epoch} loss is {epoch_loss)}") 
```
